src/Functions/initcap.cpp

```diff
@@ -62,7 +62,36 @@ using FunctionInitcap = FunctionStringToString<InitcapImpl, NameInitcap>;
 
 REGISTER_FUNCTION(Initcap)
 {
-    factory.registerFunction<FunctionInitcap>({}, FunctionFactory::Case::Insensitive);
+    FunctionDocumentation::Description description = R"(
+Converts the first letter of each word to upper case and the rest to lower case.
+Words are sequences of alphanumeric characters separated by non-alphanumeric characters.
+
+:::note
+Because `initcap` converts only the first letter of each word to upper case you may observe unexpected behaviour for words containing apostrophes or capital letters.
+This is a known behaviour and there are no plans to fix it currently.
+:::
+)";
+    FunctionDocumentation::Syntax syntax = "initcap(s)";
+    FunctionDocumentation::Arguments arguments = {
+        {"s", "Input string.", {"String"}}
+    };
+    FunctionDocumentation::ReturnedValue returned_value = {"Returns `s` with the first letter of each word converted to upper case.", {"String"}};
+    FunctionDocumentation::Examples examples = {
+    {
+        "Usage example",
+        "SELECT initcap('building for fast')",
```

Comment:
I think we could include an example where initcap() doesn't work well (:::note above): `mother''s daughter`, `joe McAdam`. But it is up to you.